src/cursor/cur_table.c

```diff
@@ -525,6 +525,13 @@ __curtable_insert(WT_CURSOR *cursor)
 	}
 
 	/*
+	 * The cursor is no longer positioned. This isn't just cosmetic, without
+	 * a reset, iteration on this cursor won't start at the beginning/end of
+	 * the table.
+	 */
+	APPLY_CG(ctable, reset);
+
+	/*
```

Comment:
I think this is only required in the `__curtable_update` case above?  There's no need to reset after inserts, right?